src/ulid.cr

```diff
@@ -19,6 +19,35 @@ module ULID
     encode_time(seed_time, TIME_LEN) + encode_random(RANDOM_LEN)
   end
 
+  class IncorrectLength < Exception; end
+
+  class InvalidChars < Exception; end
+
+  # Validate a string is a ULID
+  #
+  # ```
+  # ULID.valid?("01B3EAF48P97R8MP9WS6MHDTZ3")
+  # # => true
+  # ```
+  def valid?(ulid : String) : Bool
+    # Incorrect length
+    return false unless ulid.size == TIME_LEN + RANDOM_LEN
+    # Invalid chars
+    return false unless ulid.upcase.chars.all? &.in?(ENCODING)
+
+    return true
+  end
+
+  # Decode ULID seedtime
+  #
+  # ```
+  # ULID.seed_time("01EVDK4C0Q275A7AHHEVX02DCG")
+  # # => 2021-02-25 22:14:43.994000000 UTC
+  # ```
+  def seed_time(ulid : String) : Time
+    Time.unix_ms(ulid[0..9].to_i64(32))
```

Comment:
This is incorrect. You'll need to invert the process used in `#encode_time`. Importantly, the alphabet used for encoding is _Crockford 32_ rather than vanilla base32.

Reply:
Ah yep. I'll move this method into a separate PR.

Reply:
Added a working version back into the PR